### PR TITLE
#35 resolution of signatures and order of operations

### PR DESCRIPTION
**Pull request by spyhunter99**
ok fun times, this documents what the new procedure is publishing with signatures

1) publish with GPG unfortunately has to be a 3 step process

``` bash
$ ./gradlew clean
$ ./gradlew install -Pprofile=sources,javadoc
$ ./gradlew publish -Pprofile=sources,javadoc,sign
```

2) publish without GPG is just as simple as it was before

``` bash
$ ./gradlew clean publish -Pprofile=sources,javadoc,sign
```

3) install locally with GPG without publishing.... doesn't work

``` bash
$ ./gradlew clean install -Pprofile=sources,javadoc
$ ./gradlew install -Pprofile=sources,javadoc,sign
```

4) build, then sign

``` bash
$ ./gradlew clean install -Pprofile=sources,javadoc
# ./gradlew signArtifacts -Pprofile=sources,javadoc,sign
```

### CODE REVIEW

**Comment by spyhunter99** [comment]
just ran into an issue with this one

gradle clean install fails because it doesn't generate release artifacts
gradle clean install -Pprofile=sources,javadocs does generate release artifacts

**Comment by chrisdoyle** [comment]
Should I wait until that is addressed before reviewing it for merging?

**Comment by spyhunter99** [comment]
found the fix stand by

**Comment by spyhunter99** [comment]
fixed

File: gradle/maven-support.gradle [hunk 1]
```diff
@@ -25,11 +25,29 @@ project.ext.pgpAvailable=false;
...
+    }
+    return false;
+
+}
 
 ext.gpgPath = (
         hasProperty('GPG_PATH')
                 ? GPG_PATH
                 : "/usr/local/bin/gpg"
+    //TODO maybe a defaut path for windows?
```
Comment (chrisdoyle) [inline gradle/maven-support.gradle:50]: Good idea.  I am unable to test it as I do not have a machine running that.  :)

File: gradle/maven-support.gradle [hunk 2]
```diff
@@ -973,80 +1010,249 @@ class Container{
...
     project.ext.pgpAvailable=false;
     return false;
-}
\ No newline at end of file
+}
+
+
+/*
+
+Well now, you've made it to the end, below this line is exerpimental code.
```
Comment (chrisdoyle) [inline gradle/maven-support.gradle]: I'd go ahead and purge all of the big commented out blurbs...

**Pull request merged**